rbc/src/main/java/com/library/rbc/service/BookService.java

```diff
@@ -34,4 +39,63 @@ public BookDto addNewBook(BookDto bookDTO) {
     bookRepository.save(savedBook);
     return bookDTO;
   }
+
+  public Page<BookDto> getBooksBy(Pageable pageable,
+      List<String> bookCategories,
+      List<String> bookStatuses) {
+    Page<BookDto> resultedList;
+    if (bookCategories == null)
+      resultedList = returnBooksByStatuses(pageable, bookStatuses);
+    else {
+      resultedList = returnBooksByCategoriesAndStatuses(pageable,bookStatuses,bookCategories);
+    }
+    if (resultedList.getTotalElements() == 0) {
+      throw new BookNotFoundException("There are no books that match your request");
+    }
+    return resultedList;
+  }
```

Comment:
If there are zero books, it is ok to return an empty list in the response body. Then you can simplify further:
```java
public Page<BookDto> getBooksBy(Pageable pageable,
      List<String> bookCategories,
      List<String> bookStatuses) {
    if (bookCategories == null)
      return returnBooksByStatuses(pageable, bookStatuses);
    }
    return returnBooksByCategoriesAndStatuses(pageable,bookStatuses,bookCategories);
  }
```